server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalVariableWidthHistogram.java

```diff
@@ -422,7 +500,13 @@ private void orderBucketsByXContentKey(List<Bucket> buckets, ReduceContext reduc
     public InternalAggregation doReduce(List<InternalAggregation> aggregations, ReduceContext reduceContext) {
         List<Bucket> reducedBuckets = reduceBuckets(aggregations, reduceContext);
 
-        return new InternalVariableWidthHistogram(getName(), reducedBuckets, bucketInfo, numClusters,
+        if(reduceContext.isFinalReduce()) {
```

Comment:
This feels ok. At least, it is, like you said, what `auto_date_histogram` is doing. I think in the future we'll do more with partial reduce then we do now and we might have to change it, but for now it is final. At best you can perform the merges at each pass, but I haven't thought enough about your algorithm to have any idea if that is safe.

At least, conceptually, you are only going to get more buckets on subsequent passes. I feel like `auto_date_histogram` might have a chance to do things like bump up to a higher level of rounding. But I only am thinking about that because I was hacking on it this morning. Your aggs, I'm not sure. Not yet, at least. Sometimes it is easy to do, like `max`, right? But bucketing aggs almost always require more thought.